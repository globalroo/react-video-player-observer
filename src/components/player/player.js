import "./player.css";

import React, { Component } from "react";

export const createComponentDidMount = (context) => () => {
	context.props.onPlayerLoaded(context._player);
};

export class Player extends Component {
	componentDidMount = createComponentDidMount(this);
	render() {
		return <video controls ref={(ref) => this._player = ref} />;
	}
}
