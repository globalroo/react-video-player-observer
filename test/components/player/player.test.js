import React from "react";
import * as renderer from "react-test-renderer";
import { Player } from "src/components/player/player";

describe("Player component tests", () => {
	test("Player component renders as expected", () => {
		const spy = jest.fn();
		const component = renderer.create(
			<Player onPlayerLoaded={spy} />
		);
		expect(component.toJSON()).toMatchSnapshot();
		expect(spy).toHaveBeenCalledTimes(1);
	});
});
