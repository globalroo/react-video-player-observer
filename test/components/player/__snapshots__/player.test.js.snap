// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Player component tests Player component renders as expected 1`] = `
<video
  controls={true}
/>
`;
